lock_api/src/mutex.rs

```diff
@@ -93,6 +98,35 @@ pub struct Mutex<R: RawMutex, T: ?Sized> {
     data: UnsafeCell<T>,
 }
 
+// Copied and modified from serde
+#[cfg(feature = "enable_serde")]
+impl<R, T> Serialize for Mutex<R, T>
+where
+    R: RawMutex,
+    T: Serialize + ?Sized,
+{
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        self.lock().serialize(serializer)
+    }
+}
+
+#[cfg(feature = "enable_serde")]
+impl<'de, R, T> Deserialize<'de> for Mutex<R, T>
+where
+    R: RawMutex,
+    T: Deserialize<'de> + ?Sized
+{
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        Deserialize::deserialize(deserializer).map(|value| Mutex::new(value))
```

Comment:
You can just do `map(Mutex::new)`  here. Same for the other `Deserialize` implementations.

Reply:
Now changed.